packages/@o3r/components/builders/metadata-check/index.it.spec.ts

```diff
@@ -0,0 +1,407 @@
+        'propertyName': 'prop6'
+      },
+      'after': {
+        'libraryName': '@new/lib6',
+        'configName': 'NewConfig6',
+        'propertyName': 'newProp6Name'
+      }
+    },
+    { // Rename configuration name for all properties
+      'contentType': 'CONFIG',
+      'before': {
+        'libraryName': '@o3r/lib7',
+        'configName': 'MyConfig7'
+      },
+      'after': {
+        'libraryName': '@o3r/lib7',
+        'configName': 'NewConfig7'
+      }
+    },
+    { // Rename library name for all configurations
+      'contentType': 'CONFIG',
+      'before': {
+        'libraryName': '@o3r/lib8'
+      },
+      'after': {
+        'libraryName': '@new/lib8'
+      }
+    },
+    { // Move configuration to a new library
+      'contentType': 'CONFIG',
+      'before': {
+        'libraryName': '@o3r/lib9',
+        'configName': 'MyConfig9'
+      },
+      'after': {
+        'libraryName': '@new/lib9',
+        'configName': 'MyConfig9'
+      }
+    }
+  ]
+};
+
+const createProp = (name: string): ConfigProperty => ({
+  name,
+  type: 'string',
+  description: '',
+  label: name
+});
+
+const createConfig = (library: string, name: string, propertiesName: string[]): ComponentConfigOutput => ({
+  library,
+  name,
+  properties: propertiesName.map(createProp),
+  type: 'EXPOSED_COMPONENT',
+  path: ''
+});
+
+const previousConfigurationMetadata: ComponentConfigOutput[] = [
+  createConfig('@o3r/lib0', 'MyConfig0', ['prop0']),
+  createConfig('@o3r/lib1', 'MyConfig1', ['prop1']),
+  createConfig('@o3r/lib2', 'MyConfig2', ['prop2']),
+  createConfig('@o3r/lib3', 'MyConfig3', ['prop3']),
+  createConfig('@o3r/lib4', 'MyConfig4', ['prop4']),
+  createConfig('@o3r/lib5', 'MyConfig5', ['prop5']),
+  createConfig('@o3r/lib6', 'MyConfig6', ['prop6']),
+  createConfig('@o3r/lib7', 'MyConfig7', ['prop7']),
+  createConfig('@o3r/lib8', 'MyConfig8', ['prop8']),
+  createConfig('@o3r/lib9', 'MyConfig9', ['prop9'])
+];
+
+const newConfigurationMetadata = [
+  createConfig('@o3r/lib0', 'MyConfig0', ['prop0']),
+  createConfig('@o3r/lib1', 'MyConfig1', ['newProp1Name']),
+  createConfig('@o3r/lib2', 'NewConfig2', ['prop2']),
+  createConfig('@o3r/lib3', 'NewConfig3', ['newProp3Name']),
+  createConfig('@new/lib4', 'MyConfig4', ['prop4']),
+  createConfig('@new/lib5', 'MyConfig5', ['newProp5Name']),
+  createConfig('@new/lib6', 'NewConfig6', ['newProp6Name']),
+  createConfig('@o3r/lib7', 'NewConfig7', ['prop7']),
+  createConfig('@new/lib8', 'MyConfig8', ['prop8']),
+  createConfig('@new/lib9', 'MyConfig9', ['prop9'])
+];
+
+function writeFileAsJSON(path: string, content: object): void {
+  writeFileSync(path, JSON.stringify(content), { encoding: 'utf8' });
+}
+
+function getLatestPackageVersion(packageRef: string, execOptions: ExecSyncOptions): string | undefined {
+  try {
+    const result = packageManagerInfo(packageRef, ['--json'], execOptions);
+    const latestVersion = JSON.parse(result)['dist-tags']?.latest;
```

Comment:
this can be done without having to parse the JSON using `npm info ${packageRef} version` (see https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/test-helpers/src/prepare-test-env.ts#L76)
we could add this function in @o3r/test-helpers